Pass split shard count summary through acquireSearcher

bcully opened the pull request:
This will allow us to install search filters on shards being split according to whether the coordinating node is including new search shards in its requests or not.

bcully committed:
Pass split shard count summary through acquireSearcher

This will allow us to install search filters on shards being split according to whether
the coordinating node is including new search shards in its requests or not.

Commit:
[CI] Auto commit changes from spotless

lkts commented on server/src/main/java/org/elasticsearch/index/engine/Engine.java:
```diff
@@ -999,14 +1000,22 @@ public final SearcherSupplier acquireSearcherSupplier(Function<Searcher, Searche
 
     // Allows subclasses to wrap the DirectoryReader before it is used to create Searchers
-    protected DirectoryReader wrapDirectoryReader(DirectoryReader reader) throws IOException {
+    protected DirectoryReader wrapDirectoryReader(DirectoryReader reader, SplitShardCountSummary ignored) throws IOException {
```
This is always called from `acquireSearcherSupplier` so maybe we can introduce `AcquireSearcherRequest` and pass it down here? Looking at naming in `SearcherScope`, something like having two variant `AcquireSearcherRequest.Internal` and `AcquireSearcherRequest.External` may work. I am not sure if summary is needed for internal searchers, it feels like it won't be?

bcully committed:
Make FrozenEngine and ReadOnlyEngine implement the base acquireSearcherSupplier method

This is the one that is called by all the others. Should fix the test cases I looked at.

bcully committed:
Merge branch 'reshard-install-source-search-filter' of github.com:bcully/elasticsearch into reshard-install-source-search-filter

bcully committed:
Move wrapDirectoryReader to wrapExternalDirectoryReader

After reviewing the users of SearcherScope.INTERNAL none of them appear to
want filtering, so it is simpler and safer not to provide an interface for them to do so.

bcully committed:
Inject split shard count summary in createSearchContext

bcully committed:
Merge branch 'main' of github.com:elastic/elasticsearch into reshard-install-source-search-filter

bcully committed:
Merge branch 'main' of github.com:elastic/elasticsearch into reshard-install-source-search-filter

bcully committed:
PR feedback: rename acquireSearcherSupplier

Call the variant that takes a splitShardCountSummary acquireExternalSearcherSupplier.
